src/holder.rs

```diff
@@ -1,7 +1,79 @@
 use crate::sync::atomic::AtomicPtr;
 use crate::{Domain, HazPtrObject, HazPtrRecord};
+use std::mem::{ManuallyDrop, MaybeUninit};
 use std::sync::atomic::Ordering;
 
+pub struct HazardPointerArray<'domain, F, const N: usize> {
+    haz_ptrs: [ManuallyDrop<HazardPointer<'domain, F>>; N],
```

Comment:
Let's leave a comment here about why the `ManuallyDrop` is necessary for future reference.